dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OidcIdentityStore.java

```diff
@@ -116,14 +118,27 @@ private OpenIdContext createOpenIdContext(String subjectIdentifier, TokenRespons
         return openIdContext;
     }
 
-    private AccessToken createAccessTokenFromTokenResponse(long tokenMinValidityInMillis, TokenResponse tokenResponse) {
+    private AccessToken createAccessTokenFromTokenResponse(long tokenMinValidityInMillis, TokenResponse tokenResponse, JwtClaims jwtClaims) {
+        final String METHODNAME = "createAccessTokenFromTokenResponse";
+
         Map<String, String> tokenResponseRawMap = tokenResponse.asMap();
         long expiresIn = 0L;
         if (tokenResponseRawMap.containsKey(OpenIdConstant.EXPIRES_IN)) {
             expiresIn = Long.parseLong(tokenResponseRawMap.get(OpenIdConstant.EXPIRES_IN));
         }
 
-        // TODO: Determine if this is a JWT Access Token and use proper constructor.
+        if (tokenResponseRawMap.containsKey(JakartaSec30Constants.TOKEN_TYPE)) {
+            String token_type = tokenResponseRawMap.get(JakartaSec30Constants.TOKEN_TYPE);
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, METHODNAME, "A claimsMap exists, the " + JakartaSec30Constants.TOKEN_TYPE + " is " + token_type);
+            }
+            if (token_type != null && Type.BEARER.toString().equalsIgnoreCase(token_type)) {
```

Comment:
Do we always know that a token_type set to "Bearer" will be a JWT?

Reply:
Yeah -- I'm revamping this to check if the accessTokenString is parseable as a jwt token or not.

Reply:
Redid this portion.